### PR TITLE
Guard t_openmp_instance with KOKKOS_ENABLE_DEPRECATED_CODE_3

### PR DESCRIPTION
**Pull request by masterleinad**
Following https://github.com/kokkos/kokkos/pull/5105#discussion_r948338397, this pull request guards `t_openmp_instance` with `KOKKOS_ENABLE_DEPRECATED_CODE_3` such that it's removed when removing `partition_master`.

### CODE REVIEW

**Comment by PhilMiller** [comment]
I know I'm again jumping the gun on a draft PR, but is this also going to need to be applied to 3.7, since it's an added 3.x deprecation?

**Comment by masterleinad** [comment]
> I know I'm again jumping the gun on a draft PR, but is this also going to need to be applied to 3.7, since it's an added 3.x deprecation?

The changes here are purely internal and it's merely intended for us not to forget removing the variable as well as making sure that everything apart from `partition_master` really works without it.

**Comment by masterleinad** [comment]
Retest this please.

**Comment by masterleinad** [comment]
Only the `OPENMPTARGET-Clang` CI is timing out. This looks unrelated.

**Review by Rombur — CHANGES_REQUESTED** [review]
I would prefer that we do not merge this PR for the moment. This will create unnecessary conflicts with #5105 The `FIXME_OPENMP` are enough for now.

**Comment by crtrott** [comment]
I leave this up to Bruno if he wants this or not. I don't think its necessary there are already FIXME comments, and its not public code anyway.

**Comment by crtrott** [comment]
@Rombur what do you want to do with this?

**Comment by masterleinad** [comment]
Retest this please.

File: core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp
Comment (dalg24) [inline core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp:261]:
```suggestion
#ifdef KOKKOS_ENALBE_DEPRECATED_CODE_3
```

**Review by Rombur — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Pull request merged**
